### PR TITLE
python312Packages.kalshi-python: init at 2.0.0

### PR DESCRIPTION
**Pull request by RobbieBuxton**
Adds the [Kalshi](https://kalshi.com/) [python-api](https://github.com/Kalshi/kalshi-python) library. I have tested that the library works and can run their [api demo](https://kalshi.com/api). 

I am unsure if `kalshi-python` is the best name as it's fairly redundant in `nixpkgs` but that is the name they chose. I'm happy to change if people think otherwise.  

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix
Comment (natsukium) [inline maintainers/maintainer-list.nix:1]: Could you split this change into another commit?

File: pkgs/development/python-modules/kalshi-python/default.nix [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+  dependencies = [
+    urllib3
+    certifi
+    python-dateutil
+    six
+  ];
+
+  build-system = [ setuptools ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/kalshi-python/default.nix:31]: Could you enable tests with `pytestCheckHook` and `pythonImportsCheck`?
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#testing-python-packages-testing-python-packages

File: pkgs/development/python-modules/kalshi-python/default.nix [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  meta = {
+    description = "This library is the official python SDK for algorithmic trading on Kalshi.";
```
Comment (natsukium) [inline pkgs/development/python-modules/kalshi-python/default.nix:33]:
```suggestion
    description = "Official python SDK for algorithmic trading on Kalshi";
```
see https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

**Review by natsukium — CHANGES_REQUESTED** [review]
Since this is a python library, please change your commit message to `python312Packages.kalshi-python: init at 2.0.0`.
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#contributing-guidelines-contributing-guidelines

File: maintainers/maintainer-list.nix
Comment (RobbieBuxton) [reply maintainers/maintainer-list.nix:1]: Sure thing, to both requests!

File: pkgs/development/python-modules/kalshi-python/default.nix (hunk 1 above)
Comment (RobbieBuxton) [reply pkgs/development/python-modules/kalshi-python/default.nix:31]: I'll look into it, I suspect they will fail in the sandbox as they are testing against the API tho

**Comment by RobbieBuxton** [comment]
@natsukium Thanks for the review, hopefully I have incorporated your changes!

**Comment by ofalvai** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 352048`

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python311Packages.kalshi-python</li>
    <li>python311Packages.kalshi-python.dist</li>
    <li>python312Packages.kalshi-python</li>
    <li>python312Packages.kalshi-python.dist</li>
  </ul>
</details>

**Comment by RobbieBuxton** [comment]
@natsukium any advice on how to proceed here?

**Review by natsukium — APPROVED** [review]
Sorry for the late review. LGTM, thanks for your contribution!
